### PR TITLE
Loading C2 metadata

### PR DESCRIPTION
**Pull request by dunkgray**

### CODE REVIEW

File: tests/data.py [hunk 1]
```diff
@@ -9,9 +9,16 @@
...
+LS7_SCENERTC2 = pjoin(DATA_DIR, "LANDSAT7", "LE71140812021051EDC00__C2_RT")
 LS8_SCENE1 = pjoin(
     DATA_DIR, "LANDSAT8", "LS8_OLITIRS_OTH_P51_GALPGS01-032_090_084_20131011"
 )
+LS8_SCENE1C2 = pjoin(
+    DATA_DIR, "LANDSAT8", "LC08_L1TP_092084_20201029_20201106_02_T1_MTL.txt"
+)
+LS8_SCENE1C2 = pjoin(DATA_DIR, "LANDSAT8", "unpacked-C2")
+LS8_SCENERTC2 = pjoin(DATA_DIR, "LANDSAT8", "LC81060632021051LGN00__C2_RT")
+
```
Comment (truth-quark) [inline tests/data.py:21]: Outside the scope of the PR, but the `pathlib.Path` class makes path concatenation somewhat nicer. I'm not sure what the goal is for future changes is though.

File: tests/test_mtl.py [hunk 2]
```diff
@@ -138,6 +153,49 @@ def test_load(self):
         self.assertTrue("PROJECTION_PARAMETERS" in tree)
 
 
+class Landsat8MTL1C2ParserTest(unittest.TestCase):
+    def test_load(self):
+        tree = load_mtl(L8_MTL1C2)
+        self.assertEqual(len(tree), 10)
+        self.assertTrue("PRODUCT_CONTENTS" in tree)  # was METADATA_FILE_INFO
+        # self.assertTrue("PRODUCT_METADATA" in tree)
```
Comment (truth-quark) [inline tests/test_mtl.py]: Should this line be deleted?

File: tests/test_mtl.py [hunk 3]
```diff
@@ -138,6 +153,49 @@ def test_load(self):
...
+        self.assertTrue("IMAGE_ATTRIBUTES" in tree)  # PRODUCT_METADATA info added here
+        self.assertTrue("LEVEL1_MIN_MAX_RADIANCE" in tree)
+        self.assertTrue("LEVEL1_MIN_MAX_REFLECTANCE" in tree)
+        self.assertTrue("LEVEL1_MIN_MAX_PIXEL_VALUE" in tree)
+        self.assertTrue("LEVEL1_RADIOMETRIC_RESCALING" in tree)
+        self.assertTrue("LEVEL1_THERMAL_CONSTANTS" in tree)
+        self.assertTrue("LEVEL1_PROJECTION_PARAMETERS" in tree)
```
Comment (truth-quark) [inline tests/test_mtl.py:169]: These strings are repeated a few times. Do they exist as "constants" anywhere to protect against typos?

File: wagl/acquisition/__init__.py [hunk 4]
```diff
@@ -269,6 +287,14 @@ def worldview2_acquisitions_via_xml(pathname):
     return AcquisitionsContainer(label=pathname, granules={granule_id: res_groups})
 
 
+def get_collection_map(data_keys):
+    if "PRODUCT_METADATA" in data_keys:
+        coll = "C1"
+    else:
+        coll = "C2"
```
Comment (truth-quark) [inline wagl/acquisition/__init__.py]: How about `coll = "C1" if "PRODUCT_METADATA" in data_keys else "C2"`?

File: wagl/mtl.py [hunk 5]
```diff
@@ -74,4 +75,9 @@ def parse(lines, tree, level=0):
 
     parse(data, tree)
 
+    if root is None:
+        keys = list(tree.keys())
+        if len(keys) == 1:
```
Comment (truth-quark) [inline wagl/mtl.py:79]: Not sure of the context of L1 metadata files, but does the `if len(keys) == 0` case need to be tested?

File: tests/test_acquisition.py [hunk 6]
```diff
@@ -6,7 +6,14 @@
...
 from wagl.temperature import temperature_at_sensor
 
-from .data import LS5_SCENE1, LS7_SCENE1, LS8_SCENE1
+from .data import (
+    LS5_SCENE1,
+    LS7_SCENE1,
+    LS8_SCENE1,
+    LS8_SCENE1C2,
+    LS8_SCENERTC2,
+    LS7_SCENERTC2,
```
Comment (truth-quark) [inline tests/test_acquisition.py:15]: Do these constants need a scene number in the name? e.g. `LS8_SCENE1RTC2`

File: tests/test_acquisition.py [hunk 7]
```diff
@@ -246,6 +308,107 @@ def test_tzinfo(self):
...
+
+    def test_sensor_id(self):
+        for acq in self.acqs:
+            self.assertEqual(acq.sensor_id, "ETM+")
+
+    def test_platform_id(self):
+        for acq in self.acqs:
+            self.assertEqual(acq.platform_id, "LANDSAT_7")
+
+    def test_samples(self):
```
Comment (truth-quark) [inline tests/test_acquisition.py:368]: Is this testing the number or count of samples?

File: tests/test_acquisition.py [hunk 8]
```diff
@@ -246,6 +308,107 @@ def test_tzinfo(self):
...
+    def test_read(self):
+        self.assertEqual(self.acqs[0].data()[40, 30], 50)
+
+    def test_spectral_filter_cfg_vsir(self):
+        self.assertEqual(self.acqs[0].spectral_filter_name, "landsat7_vsir.flt")
+
+    def test_spectral_filter_cfg_thermal(self):
+        self.assertEqual(self.acqs[5].spectral_filter_name, "landsat7_thermal.flt")
+
+    def test_temperature61(self):
```
Comment (truth-quark) [inline tests/test_acquisition.py:383]: Does `61` have some contextual meaning? (and 62 below?)

**Review by truth-quark — CHANGES_REQUESTED** [review]
Classifying this as "Request changes" status, mostly due to unfamiliarity with the `wagl` project.

* This PR includes multiple binary files and XML data. Is there a known reason why binaries are being added to the code repo instead of a separate, versioned data store?
* There's also direct I/O in the tests: loading files for data. Is this approach a legacy from `wagl`'s dev history?
* test_acquisitions.py looks ok, but I’m lacking the right background context to know if anything is wring or missing. Is there a second reviewer around?

File: tests/test_acquisition.py (hunk 8 above)
Comment (dunkgray) [reply tests/test_acquisition.py:383]: Good question. Not that I'm aware of.  This test is based on the Landsat7 Collection 1 test.

File: tests/test_acquisition.py (hunk 7 above)
Comment (dunkgray) [reply tests/test_acquisition.py:368]: I'll change the test name.  It is actually a width value

File: tests/test_acquisition.py (hunk 6 above)
Comment (dunkgray) [reply tests/test_acquisition.py:15]: I dropped the 1.  I didn't think it was adding any value.

File: wagl/mtl.py (hunk 5 above)
Comment (dunkgray) [reply wagl/mtl.py:79]: That could be done.  The updated code and the old code will return a key error here `tree[root]`

File: wagl/acquisition/__init__.py (hunk 4 above)
Comment (dunkgray) [reply wagl/acquisition/__init__.py]: ok, thanks

File: tests/test_mtl.py (hunk 3 above)
Comment (dunkgray) [reply tests/test_mtl.py:169]: Good point.  I don't think they do exist as constants.  They could in the future.  For these changes I think I'll start with the current structure.

File: tests/test_mtl.py (hunk 2 above)
Comment (dunkgray) [reply tests/test_mtl.py]: Yes, thanks, done.  And I've added more context to the comments.

File: tests/data.py (hunk 1 above)
Comment (dunkgray) [reply tests/data.py:21]: Yes, this could be done in the future.

**Comment by dunkgray** [comment]
>Classifying this as "Request changes" status, mostly due to unfamiliarity with the wagl project.

>This PR includes multiple binary files and XML data. Is there a known reason why binaries are being added to the code repo >instead of a separate, versioned data store?
Using a separate, versioned data store is an option too. The current version of wagl uses small binaries as test data.  I decided to use the current structure.   

>There's also direct I/O in the tests: loading files for data. Is this approach a legacy from wagl's dev history?
Yes, I'm testing the collection 2 data using the same way done for collection 1.

>test_acquisitions.py looks ok, but I’m lacking the right background context to know if anything is wring or missing. Is there a >second reviewer around?
That's ok.  I'll get a second reviewer.

Thanks Ban

File: tests/test_acquisition.py (hunk 8 above)
Comment (sixy6e) [reply tests/test_acquisition.py:383]: So 61 and 62 are the band labels from USGS for Landsat-7.

File: tests/data.py (hunk 1 above)
Comment (sixy6e) [reply tests/data.py:21]: Was briefly discussed a long time ago, but was relegated to a `nice to have` whilst algorithmic generalisation and cleanup took place.
But yes, could be resolved in a separate PR.

File: tests/test_acquisition.py (hunk 6 above)
Comment (sixy6e) [reply tests/test_acquisition.py:15]: Yeah, they could be dropped.  From memory, it was to support more than 1 acquisition from a given sensor, such as multiple versions.  But over time we've decided to stop supporting older versions of `scenes` that USGS would create as it was adding unnecessary complexity.  By older versions, I mean metadata, filenames, metadata keys are completely different, and we would attempt to support each version.  Not worth the hassle anymore.

File: tests/test_acquisition.py (hunk 7 above)
Comment (sixy6e) [reply tests/test_acquisition.py:368]: Traditionally, the sensors were acquiring `samples` across a given scanline.  Hence the term `samples`.  And the USGS supplied metadata would also use the terms `lines` and `samples`.  I don't mind either way, whatever makes it easier for someone to maintain.

File: wagl/acquisition/__init__.py [hunk 9]
```diff
@@ -35,6 +35,22 @@
 # resolution group format
 RESG_FMT = "RES-GROUP-{}"
 
+LANDSATMTLMAP = {
```
Comment (sixy6e) [inline wagl/acquisition/__init__.py:38]: Not a big fan of the condensed wording, but ok to move on with it.  As long as people are aware that there are 3 pieces to it:
* LANDSAT
* MTL
* MAP

**Review by sixy6e — APPROVED** [review]
Looks good to me.  My in-line comments are mostly for additional context.

File: tests/test_mtl.py (hunk 3 above)
Comment (sixy6e) [reply tests/test_mtl.py:169]: Yeah, don't worry about all the cleanups that could occur for this repo (there are lots, but pick and choose the battles :smile:  )

File: wagl/acquisition/__init__.py (hunk 4 above)
Comment (sixy6e) [reply wagl/acquisition/__init__.py]: Down the line, it could be removed entirely (if we continue with dropping support for previous versions).

**Comment by sixy6e** [comment]
Oh and for additional context.  The Travis builds are failing because the Ubuntu-GIS package that is retrieved doesn't have GDAL-3 yet.  wagl requires GDAL-3, and ideally, we transition over to using GitHub Actions, but the time to do so was lacking.

**Comment by dunkgray** [comment]
Thanks Josh

**Pull request merged**